astropy/modeling/core.py

```diff
@@ -3619,11 +3619,14 @@ def fix_inputs(modelinstance, values):
 def custom_model(*args, fit_deriv=None, **kwargs):
     """
     Create a model from a user defined function. The inputs and parameters of
-    the model will be inferred from the arguments of the function.
+    the model will be inferred from the arguments of the function. The model
+    always has one output.
```

Comment:
> The model always has one output.

I haven't looked at or used this code in years. Is this really a true statement?
I can create a model returning two outputs though the function is not well defined in that respect. 
I think we should fix it (in a different PR). 
Perhaps just remove this statement for now?

Reply:
> I can create a model returning two outputs though the function is not well defined in that respect.

By passing a `func` that is not a mathematical function, i.e. multi-valued? If that is not checked, that statement would indeed likely cause more confusion. Although further down under "Parameters" `func` is also defined as
> It must return the value of the model (typically as
        an array, but can also be a scalar for scalar inputs).

I think that also makes it clearer than the "one output".

Reply:
The example I tried is

```
@custom_model
def m(x, y, n_outputs=2):
    return x+1, y+1

m()(1,1)
Out[66]: (2.0, 2.0)
```

Though this works `custom_model` needs to be modified to not create a `Parameter` from the `n_outputs` keyword.

Reply:
```
m.n_outputs
Out[72]: Parameter('n_outputs', value=2.0)
```

Reply:
I have removed this line for this PR.

In your case above I think you are actually overriding the `n_outputs` property with a parameter (which should probably error somewhere?), and the number of outputs is really a single tuple instead of two numbers, but I think this point needs investigation beyond this PR.

Reply:
I've opened an issue at https://github.com/astropy/astropy/issues/11791 to discuss the code brought up above.